pkg/cloudprovider/aws/pricing.go

```diff
@@ -383,11 +419,29 @@ func (p *PricingProvider) updateSpotPricing(ctx context.Context) error {
 	}
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	p.spotPrices = map[string]float64{}
-	for k, v := range prices {
-		p.spotPrices[k] = v.price
+	p.spotPrices = make(map[string]map[string]float64)
+	for it, zoneData := range prices {
+		if _, ok := p.spotPrices[it]; !ok {
+			p.spotPrices[it] = make(map[string]float64)
+		}
+		for zone, data := range zoneData {
+			p.spotPrices[it][zone] = data.price
+		}
+		totalOfferings += len(zoneData)
 	}
+
 	p.spotUpdateTime = time.Now()
-	logging.FromContext(ctx).Infof("updated spot pricing with %d instance types", len(p.spotPrices))
+	logging.FromContext(ctx).Infof("updated spot pricing with %d instance types and %d offerings", len(p.spotPrices), totalOfferings)
 	return nil
 }
+
+// initialSpotPricing creates spot zonal pricing information based off of the
+// initial on-demand pricing information coming at the regional level
+func initialSpotPricing(odPricing map[string]float64) map[string]map[string]float64 {
+	m := make(map[string]map[string]float64)
```

Comment:
I think it reads a little easier using the shorthand instantiation syntax (we tend to use this elsewhere in the codebase too).

```suggestion
	m := map[string]map[string]float64{}
```